Mathlib/GroupTheory/GroupAction/Defs.lean

```diff
@@ -1128,6 +1128,12 @@ theorem Function.End.smul_def (f : Function.End α) (a : α) : f • a = f a :=
   rfl
 #align function.End.smul_def Function.End.smul_def
 
+theorem Function.End.mul_def (f g : Function.End α) : (f * g) = f ∘ g :=
```

Comment:
This feels like defeq abuse to me; I'd be happier with something like
```suggestion
theorem Function.End.mul_def (f g : Function.End α) : (f * g) = toEnd (f ∘ g) :=
```

Reply:
I'd be happy with a TODO comment instead, as long as these aren't simp lemmas

Reply:
I'm not sure exactly what is TODO. There are a few ways `toEnd` could be done, we could completely forbid using `End` as a function without using a `FunLike` instance, or be less strict.

Reply:
I added some comments